app/gateway-oss/2.4.x/proxy.md

```diff
@@ -695,6 +707,38 @@ Host: ...
 
 [Back to top](#introduction)
 
+#### Normalization behavior
+
+In order to prevent trivial Route match bypass, the incoming request URI from client
+is always normalized according to [RFC 3986](https://tools.ietf.org/html/rfc3986)
+before router match occurs. Specifically, the following normalization techniques are
+used for incoming request URIs which are selected because they generally do not change
+semantics of the request URI:
+
+1. Percent-encoded triplets will be converted to uppercase. e.g. `/foo%3a` will become `/foo%3A`.
+2. Percent-encoded triplets of unreserved characters will be decoded. e.g. `/fo%6F` will become `/foo`.
+3. Dot-segments are removed as necessary. e.g. `/foo/./bar/../baz` will become `/foo/baz`.
+4. Duplicate slashes are merged. e.e. `/foo//bar` will become `/foo/bar`.
+
+`paths` attribute of the Route object are also normalized. It is achieved by first determining
+if the path is a plain text or regex path. Based on the result, different normalization techniques
+are used:
+
+For plain text Route path:
+
+Same normalization technique as above is used, that is, method 1 through 4.
```

Comment:
```suggestion
Same normalization technique as above is used, that is, methods 1 through 4.
```

Should this really be called "methods"? Perhaps the items 1-4 should have a separate heading, such as Normalization or URI normalization.